tests/scaletest/lib/ingress.go

```diff
@@ -0,0 +1,349 @@
+package lib
+
+import (
+	"flag"
+	"path/filepath"
+	"strconv"
+	"testing"
+	coreV1 "k8s.io/api/core/v1"
+	appsV1 "k8s.io/api/apps/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+)
+
+var ingressResource = schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "ingresses"}
+var kubeClient dynamic.Interface
+var coreV1Client corev1.CoreV1Interface
+var appsV1Client appsv1.AppsV1Interface
+
+func CreateApp(appName string, namespace string){
+	deploymentSpec := &appsV1.Deployment{
+		ObjectMeta: metaV1.ObjectMeta{
+			Name: appName,
+			Namespace: namespace,
+		},
+		Spec: appsV1.DeploymentSpec{
+			ProgressDeadlineSeconds: func() *int32 { i := int32(600); return &i }(),
+			RevisionHistoryLimit: func() *int32 { i := int32(10); return &i }(),
+			Replicas: func() *int32 { i := int32(2); return &i }(),
+			Selector: &metaV1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": appName,
+				},
+			},
+			Template: coreV1.PodTemplateSpec{
+				ObjectMeta: metaV1.ObjectMeta{
+					Labels: map[string]string{
+						"app": appName,
+					},
+				},
+				Spec: coreV1.PodSpec{
+					Containers: []coreV1.Container{
+						{
+							Name:  appName,
+							Image: "avinetworks/server-os",
+							ImagePullPolicy: func() coreV1.PullPolicy { str  := coreV1.PullPolicy("IfNotPresent"); return str}(),
+							Ports: []coreV1.ContainerPort{
+								{
+									Name:          "http",
+									Protocol:      coreV1.ProtocolTCP,
+									ContainerPort: 8080,
+								},
+							},
+							TerminationMessagePath: "/dev/termination-log",
+							TerminationMessagePolicy: func() coreV1.TerminationMessagePolicy { str := coreV1.TerminationMessagePolicy("File"); return str}(),
+						},
+					},
+					DNSPolicy: func() coreV1.DNSPolicy{ str := coreV1.DNSPolicy("ClusterFirst"); return str}(),
+					RestartPolicy: func() coreV1.RestartPolicy { str := coreV1.RestartPolicy("Always"); return str}(),
+					SchedulerName: "default-scheduler",
+					TerminationGracePeriodSeconds: func() *int64 { i := int64(30); return &i }(),
+				},
+			},
+		},
+	}
+
+	_, err := appsV1Client.Deployments(namespace).Create(deploymentSpec)
+	if err != nil {
+		panic(err)
```

Comment:
Panic would crash the entire program I think. Instead, we can only fail the current Testcase, print error logs and try to continue with next set of tests.

Reply:
@monotosh-avi If the deployment isn't getting deployed, do we continue with other tests? I guess it would depend on whether the tests are self contained. But yeah, for other cases, `panic` isn't required.

Reply:
@hemantavi If the failure is due to a name conflict where the deployment already exists, next test case can be executed If the app name is different.

Reply:
Sure, makes sense. @monotosh-avi

Reply:
The same deployment is used for all the test cases. Also since the prefix for this deployment name is being given by the tester in the testbed file, a conflict will be a rare case.